### DISCUSSION TITLE
Questions about direct solvers

### DISCUSSION THREAD

**Discussion by zhyxue0** [firedrakeproject/firedrake#2692] [Firedrake support]
Hello everyone. I am solving a time-dependent problem, but the assembled matrix  A is a constant coefficient matrix. In order to reduce computation time, I don't want to solve Ax = b every time, but calculate the inverse A_inv of A in the first step. 
Then at each step x = A_inv * b, how to achieve this?

**Comment by dham** [comment]
Actually storing `Ainv` is a bad idea because it will be dense and hence huge. Factorising `A` into `L` and `U` and applying the inverse using them repeatedly is a good idea. LU is the default solver and I think if you pass `constant_jacobian=True` to the `LinearVariationalProblem` then the factorisation will be reused, which is what you want.
